src/CommonLib/LDAPUtils.cs

```diff
@@ -1078,6 +1102,11 @@ public ActiveDirectorySecurityDescriptor MakeSecurityDescriptor()
 
         public string BuildLdapPath(string dnPath, string domainName)
         {
+            //Check our cached info for a fast check
+            if (CachedDomainInfo.TryGetValue(domainName, out var info))
+            {
+                return $"{dnPath},{info.DomainSearchBase}";
+            }
             var domain = GetDomain(domainName)?.Name;
             if (domain == null)
```

Comment:
This feels a little awkward since the return type of this function is `string`. Why return `null` here?